server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java

```diff
@@ -200,6 +200,48 @@ private static class OBMySQLRelationFactorVisitor extends OBParserBaseVisitor<Re
 
         private final Set<DBSchemaIdentity> identities = new HashSet<>();
 
+        @Override
+        public RelationFactor visitDrop_function_stmt(
+                com.oceanbase.tools.sqlparser.obmysql.OBParser.Drop_function_stmtContext ctx) {
+            RelationFactor relationFactor = MySQLFromReferenceFactory.getRelationFactor(
+                    ctx.relation_factor().normal_relation_factor());
+            if (relationFactor.getSchema() != null) {
+                identities.add(new DBSchemaIdentity(relationFactor.getSchema(), null));
+            }
+            return null;
+        }
+
+        @Override
+        public RelationFactor visitDrop_procedure_stmt(
+                com.oceanbase.tools.sqlparser.obmysql.OBParser.Drop_procedure_stmtContext ctx) {
+            RelationFactor relationFactor = MySQLFromReferenceFactory.getRelationFactor(
+                    ctx.relation_factor().normal_relation_factor());
+            if (relationFactor.getSchema() != null) {
+                identities.add(new DBSchemaIdentity(relationFactor.getSchema(), null));
+            }
+            return null;
+        }
+
+        @Override
+        public RelationFactor visitDrop_trigger_stmt(
+                com.oceanbase.tools.sqlparser.obmysql.OBParser.Drop_trigger_stmtContext ctx) {
+            RelationFactor relationFactor = MySQLFromReferenceFactory.getRelationFactor(
+                    ctx.relation_factor().normal_relation_factor());
+            if (relationFactor.getSchema() != null) {
+                identities.add(new DBSchemaIdentity(relationFactor.getSchema(), null));
+            }
+            return null;
+        }
+
+        @Override
+        public RelationFactor visitDrop_view_stmt(
+                com.oceanbase.tools.sqlparser.obmysql.OBParser.Drop_view_stmtContext ctx) {
+            ctx.table_list().relation_factor().stream()
```

Comment:
what about null value?

Reply:
remove it. view should be added in the permission control